Makefile.am

```diff
@@ -1,151 +0,0 @@
-
-## -----------
-## Makefile.am
-## -----------
-## Please keep this file well-commented.
-
-include $(top_srcdir)/glib-tap.mk
-
-# Our Autoconf macros are kept in the m4/ directory
-ACLOCAL_AMFLAGS = -I m4
-
-# Extra files to distribute in the tarball
-EXTRA_DIST += \
-	glib-tap.mk \
-	README.md \
-	tap-driver.sh \
-	tap-test \
-	$(NULL)
-
-# Compiler flags
-AM_CFLAGS = @WARN_CFLAGS@
-AM_LDFLAGS = @WARN_LDFLAGS@
-
-# Enable strict compiler flags and install systemd unit files to the specified
-# directory when doing 'make distcheck'.
-AM_DISTCHECK_CONFIGURE_FLAGS = \
-	--enable-compile-warnings=error \
-	--with-systemdsystemunitdir='$${prefix}/lib/systemd/system' \
-	$(NULL)
-if EOS_ENABLE_COVERAGE
-AM_DISTCHECK_CONFIGURE_FLAGS += --enable-coverage --with-coverage-dir=@EOS_COVERAGE_DIR@
-endif EOS_ENABLE_COVERAGE
-
-# Generated files that 'make clean' should erase
-CLEANFILES +=
-DISTCLEANFILES +=
-
-# Generated sources that should be built before other sources
-BUILT_SOURCES +=
-
-# Binaries that should be produced but not installed.
-noinst_PROGRAMS +=
-
-# Libraries that should be produced but not installed.
-noinst_LTLIBRARIES +=
-
-# Binaries that should be installed in libexecdir
-libexec_PROGRAMS =
-
-substitute_libexecdir = sed \
-	-e 's|@libexecdir[@]|$(libexecdir)|g' \
-	$(NULL)
-
-# dbus services
-dbusservicedir = ${datadir}/dbus-1/system-services
-dbusservice_DATA =
-
-# dbus security policy
-systembussecuritypolicydir = ${sysconfdir}/dbus-1/system.d
-dist_systembussecuritypolicy_DATA =
-
-# policykit
-policydir = ${datadir}/polkit-1/actions
-dist_policy_DATA =
-
-policyrulesdir = $(datadir)/polkit-1/rules.d
-dist_policyrules_DATA =
-
-persistentcachedir=$(localstatedir)/cache/metrics/v3
-configdir=$(sysconfdir)/metrics/
-permissions_file=$(configdir)eos-metrics-permissions.conf
-
-# systemd
-systemdsystemunit_DATA =
-
-# chown and chgrp are not allowed unless you are root or under some specific
-# other conditions that may not be met every time you run 'make'.
-install-data-hook:
-	$(MKDIR_P) --mode=g+s,ug+rwx,a+rx $(DESTDIR)$(persistentcachedir) $(DESTDIR)$(configdir)
-	chmod --recursive g+s,ug+rwx,a+rx $(DESTDIR)$(persistentcachedir) $(DESTDIR)$(configdir)
-	@if id -u metrics 2> /dev/null; then \
-		if ! chown --recursive metrics:metrics $(DESTDIR)$(persistentcachedir) $(DESTDIR)$(configdir); then \
-			echo "*************************************"; \
-			echo "Make sure to recursively change the owner and"; \
-			echo "group of the $(persistentcachedir) and the" ; \
-			echo "$(configdir) directories to 'metrics'."; \
-			echo "*************************************"; \
-		fi; \
-	else \
-		echo "*******************************************"; \
-		echo "There is no 'metrics' user on your system."; \
-		echo "If this is an EndlessOS system, upgrade the"; \
-		echo "'base-passwd' package."; \
-		echo "Otherwise, create a 'metrics' user."; \
-		echo "*******************************************"; \
-	fi
```

Comment:
Did not bother to try to preserve this since it is done by the tmpfiles snippet.

Reply:
I assume those directories simply don't need to exist at all in the `ninja install` route?

e.g. if created in the Makefile, you'd end up with distro packages actually listing an empty directory here in the package manifest, and then uninstallation might try to remove it.

Meson can do this too, if you really want it, via install_emptydir (you'd need to bump your minimum meson from 0.52.0 to 0.60.0 though).

Reply:
They don't need to exist. They are created at boot time by `systemd-tmpfiles`. (Actually on Endless OS any files placed in `/var` by packages at install time are completely ignored, so the empty directories are purely a waste of inodes.)